<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWidget</class>
 <widget class="QWidget" name="MainWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>自绘列表示例</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0,0,0">
   <item>
    <widget class="CustomListWidget" name="listWidget" native="true"/>
   </item>
   <item>
    <widget class="QWidget" name="controlPanel" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="addButton">
        <property name="text">
         <string>添加项</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="removeButton">
        <property name="text">
         <string>删除选中项</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>清空</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="selectionModeLabel">
        <property name="text">
         <string>选择模式:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="selectionModeCombo"/>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="refreshGroupBox">
     <property name="title">
      <string>实时刷新控制</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QCheckBox" name="randomRefreshCheck">
        <property name="text">
         <string>随机内容</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="refreshIntervalLabel">
        <property name="text">
         <string>刷新间隔(ms):</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="refreshIntervalSpin">
        <property name="minimum">
         <number>10</number>
        </property>
        <property name="maximum">
         <number>5000</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startStopRefreshButton">
        <property name="text">
         <string>开始刷新</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="statusGroupBox">
     <property name="title">
      <string>列表状态</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="text">
         <string>项目数: 0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="currentItemLabel">
        <property name="text">
         <string>当前选中: 无</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomListWidget</class>
   <extends>QWidget</extends>
   <header>customlistwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
